### PR TITLE
Alloy integration 

### PR DESCRIPTION
**Pull request by struan**

### CODE REVIEW

File: t/open311/endpoint/alloy.t [hunk 1]
```diff
@@ -0,0 +1,570 @@
...
+subtest "create basic problem" => sub {
+    set_fixed_time('2014-01-01T12:00:00Z');
+    my $res = $endpoint->run_test_request( 
+        POST => '/requests.json', 
+        jurisdiction_id => 'dummy',
+        api_key => 'test',
+        service_code => 'Kerbs_Missing',
+        address_string => '22 Acacia Avenue',
+        first_name => 'Bob',
+        last_name => 'Mould',
```
Comment (davea) [inline t/open311/endpoint/alloy.t]: 🤘🏼

File: perllib/Open311/Endpoint/Integration/Alloy.pm [hunk 2]
```diff
@@ -0,0 +1,655 @@
...
+            next unless $parent->{actualParentSourceTypeId} == $self->config->{defect_inspection_parent_id}; # request for service
+
+            $has_enquiry_parent = 1;
+        }
+
+        next if $has_enquiry_parent;
+
+        my $category = $self->get_defect_category( $request );
+        unless ($category) {
+            warn "No category found for defect $request->{resourceId}, source type $request->{sourceTypeId} in " . $self->jurisdiction_id . "\n";
```
[the remark this answers is not in the document]
Comment (struan) [reply perllib/Open311/Endpoint/Integration/Alloy.pm]: It is intentional in that I wanted to check that this functionality is working.

File: perllib/Open311/Endpoint/Integration/Alloy.pm [hunk 3]
```diff
@@ -0,0 +1,655 @@
...
+    my ($self, $lon, $lat) = @_;
+
+    my $point = $self->alloy->api_call("projection/point", {
+        x => $lon,
+        y => $lat,
+        dstCode => "4326",
+        srcCode => "900913",
+    });
+
+    return [ $point->{y}, $point->{x} ];
```
[the remark this answers is not in the document]
Comment (struan) [reply perllib/Open311/Endpoint/Integration/Alloy.pm]: No, geoJSON does `[x, y]` and our request class uses `[y, x]` and `reproject_coordinates` goes into geoJSON and `deproject_coordinates` goes in to our request class. I have no idea why our service class does it that way round.

File: cpanfile.snapshot [hunk 4]
```diff
@@ -104,6 +104,12 @@ DISTRIBUTIONS
       perl 5.008001
       strict 0
       warnings 0
+  Class-Data-Inheritable-0.08
```
[the remark this answers is not in the document]
Comment (struan) [reply cpanfile.snapshot]: No, this was a missing dependency of one of the other modules in cpanfile (Log::Dispatch IIRC).

**Comment by dracos** [comment]
"Not sure what the Perl 5.14 failures on Travis are about." - well:
```
    "Highway Bridges": {
      # resourceId is the value to use for the category attribute
      "resourceId": 10000000
      "Damaged/Unsafe": "Structures"
    },
```
is not valid YAML, so question might be why does it pass on the others! (It still shows the same error lots)

File: perllib/Open311/Endpoint/Integration/Alloy.pm (hunk 3 above)
[the remark this answers is not in the document]
Comment (dracos) [reply perllib/Open311/Endpoint/Integration/Alloy.pm]: I guess it's called `latlong` so is that way round. Could swap if it makes things easier.

**Pull request merged**